### PR TITLE
test(community): add a test that deletes a community category

### LINKED ISSUE #6750: Automation: Can delete a community category

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #6750

### What does the PR do

Add squish test case for deleting community category

### CODE REVIEW

File: test/ui-test/src/screens/StatusCommunityScreen.py [hunk 1]
```diff
@@ -146,12 +149,28 @@ def create_community_category(self, community_category_name, community_channel_n
             test.fail("Can't find channel " + community_channel_name)
 
         click_obj_by_name(CreateOrEditCommunityCategoryPopup.COMMUNITY_CATEGORY_BUTTON.value)
+        
+    def delete_community_category(self, community_category_name):
+        [loaded, category] = self._find_category_in_chat(community_category_name)
+        if not loaded:
+            test.fail("Can't find category " + community_category_name)
```
Comment (noeliaSD) [inline test/ui-test/src/screens/StatusCommunityScreen.py:156]: Nowadays we are using `SquishDriverVerification` and not direct calls to squish api. Could you replace it accordingly?

File: test/ui-test/src/screens/StatusCommunityScreen.py [hunk 2]
```diff
@@ -146,12 +149,28 @@ def create_community_category(self, community_category_name, community_channel_n
...
+
+        # For some reason it clicks on a first channel in category instead of category
+        squish.mouseClick(category.parent, squish.Qt.RightButton)
+        click_obj_by_name(CommunityScreenComponents.COMMUNITY_DELETE_CATEGORY_MENU_ITEM.value)
+        click_obj_by_name(CommunityScreenComponents.COMMUNITY_CONFIRM_DELETE_CATEGORY_BUTTON.value)
+        time.sleep(0.5)
```
Comment (noeliaSD) [inline test/ui-test/src/screens/StatusCommunityScreen.py:162]: Why is this needed?

File: test/ui-test/src/screens/StatusCommunityScreen.py [hunk 3]
```diff
@@ -146,12 +149,28 @@ def create_community_category(self, community_category_name, community_channel_n
...
 
     def verify_category_name(self, community_category_name):
         [result, _] = self._find_category_in_chat(community_category_name)
         if not result:
             test.fail("Can't find category " + community_category_name)
```
Comment (noeliaSD) [inline test/ui-test/src/screens/StatusCommunityScreen.py:167]: Same comment than before, use `SquishDriverVerification` here.

File: test/ui-test/src/screens/StatusCommunityScreen.py [hunk 4]
```diff
@@ -146,12 +149,28 @@ def create_community_category(self, community_category_name, community_channel_n
...
 
+    def verify_category_name_missing(self, community_category_name):
+        [result, _] = self._find_category_in_chat(community_category_name)
+        if result:
+            test.fail("Category " + community_category_name + " still exist")
```
Comment (noeliaSD) [inline test/ui-test/src/screens/StatusCommunityScreen.py:172]: Same comment than before, use `SquishDriverVerification` here.

File: test/ui-test/testSuites/suite_status/shared/scripts/sections/community_names.py [hunk 5]
```diff
@@ -8,6 +8,9 @@
 mainWindow_createChannelOrCategoryBtn_StatusBaseText = {"container": statusDesktop_mainWindow, "objectName": "createChannelOrCategoryBtn", "type": "StatusBaseText", "visible": True}
 create_channel_StatusMenuItemDelegate = {"checkable": False, "container": statusDesktop_mainWindow_overlay, "enabled": True, "objectName": "createCommunityChannelBtn", "type": "StatusMenuItemDelegate", "visible": True}
 create_category_StatusMenuItemDelegate = {"checkable": False, "container": statusDesktop_mainWindow_overlay, "enabled": True, "objectName": "createCommunityCategoryBtn", "type": "StatusMenuItemDelegate", "visible": True}
+edit_сategory_StatusMenuItemDelegate = {"checkable": False, "container": statusDesktop_mainWindow_overlay, "enabled": True, "objectName": "editCategoryMenuItem", "type": "StatusMenuItemDelegate", "visible": True}
```
Comment (noeliaSD) [inline test/ui-test/testSuites/suite_status/shared/scripts/sections/community_names.py:11]: Is `checkable` property mandatory here?

File: test/ui-test/testSuites/suite_status/tst_communityFlows/test.feature [hunk 6]
```diff
@@ -77,6 +77,17 @@ Feature: Status Desktop community
             | test-channel-1            | test-category-1             | bottom_menu      |
             | test-channel-2            | test-category-2             | right_click_menu |
 
+
+    Scenario: Admin deletes a community category
+        When the user creates a community named myCommunity, with description My community description, intro Community Intro and outro Community Outro
+        Then the user lands on the community named myCommunity
+        When the admin creates a community channel named test-channel, with description My description with the method bottom_menu
+        When the admin creates a community category named test-category, with channel test-channel and with the method bottom_menu
```
Comment (noeliaSD) [inline test/ui-test/testSuites/suite_status/tst_communityFlows/test.feature:85]:
```suggestion
        And the admin creates a community category named test-category, with channel test-channel and with the method bottom_menu
```

**Review by noeliaSD — COMMENTED** [review]
Nice job!

Some minor comments added.

File: test/ui-test/src/screens/StatusCommunityScreen.py (hunk 2 above)
Comment (MishkaRogachev) [reply test/ui-test/src/screens/StatusCommunityScreen.py:162]: to ensure that categories list is updated, on the next step we trying to find deleted category

File: test/ui-test/testSuites/suite_status/shared/scripts/sections/community_names.py (hunk 5 above)
Comment (MishkaRogachev) [reply test/ui-test/testSuites/suite_status/shared/scripts/sections/community_names.py:11]: don't think so, i just trying to keep code close other cases. It's also not necessary for categories

File: test/ui-test/testSuites/suite_status/tst_communityFlows/test.feature (hunk 6 above)
Comment (MishkaRogachev) [reply test/ui-test/testSuites/suite_status/tst_communityFlows/test.feature:85]: fixed

File: test/ui-test/testSuites/suite_status/shared/scripts/sections/community_names.py (hunk 5 above)
Comment (jrainville) [reply test/ui-test/testSuites/suite_status/shared/scripts/sections/community_names.py:11]: the names files are still a bit messy. The clean ups helped a lot, but they can still be improved,  so don,t take anything you see in those files as perfect practices :sweat_smile: 
Anyway, if we don't need `checkable`, better to remove it, that way, there is less chance for the name to break if the UI changes

**Review by jrainville — APPROVED** [review]
Good job. If you can, update the name before merging

File: test/ui-test/src/screens/StatusCommunityScreen.py (hunk 2 above)
Comment (MishkaRogachev) [reply test/ui-test/src/screens/StatusCommunityScreen.py]: removed

**Review by noeliaSD — APPROVED** [review]
LGTM! Thanks for the updates! :)
